### ISSUE TITLE
Train on logits instead of one hot vectors

### ISSUE THREAD

**Issue by Theodotus1243** [huggingface/transformers#31768]
### Feature request

Add the ability to train on Bigger LLM logits instead of one hot vectors created from tokenized input example

### Motivation

As stated on [gemma2 paper](https://storage.googleapis.com/deepmind-media/gemma/gemma-2-report.pdf) train smaller models on logits of bigger models (distilation) produces better result

### Your contribution

Details from gemma2 [here](https://storage.googleapis.com/deepmind-media/gemma/gemma-2-report.pdf)
